R/length-params.R

```diff
@@ -74,51 +68,44 @@ get_catch_types <- function(log_threshold = logger::DEBUG) {
 #' @param country_code The code of the country associated to FishBase database, default
 #' code include Timor-Leste and Indonesia. Full list at
 #' \url{https://www.fishbase.se/country/ListOfCountryCodes.php}.
-#' @param get_one Logical, indicating wether the user wants to get length data
-#' from the most recorded species.
 #'
 #' @return A data frame containing species length parameters
 #' @export
 #'
 #' @examples
 #' get_fish_length(
 #'   taxa = "Sardine", rank = "comm_name",
-#'   country_code = 626, get_one = FALSE
+#'   country_code = 626
 #' )
 get_fish_length <- function(taxa,
                             rank = NULL,
-                            country_code = NULL,
-                            get_one = NULL) {
-  take_one <- function(specs_length) {
-    specs_length %>%
-      dplyr::arrange(dplyr::desc(.data$CoeffDetermination)) %>%
-      dplyr::slice(1)
-  }
-
+                            country_code = NULL) {
   if (rank == "comm_name") {
     sp_list <- rfishbase::common_to_sci(taxa) %>%
       magrittr::extract2("Species")
   } else {
     sp_list <- list(taxa) %>%
       rlang::set_names(rank) %>%
-      do.call(rfishbase::species_list,.)
+      do.call(rfishbase::species_list, .)
+  }
+
+  if (taxa == "Abudefduf vaigiensis" | taxa == "Hyporhamphus quoyi") {
+    specs_length <- rfishbase::length_weight(taxa)
+    return(specs_length)
```

Comment:
Just curious about why there is particular treatment for these two species?

Reply:
The problem with these species was that they hadn't a country code assigned in the table returned by `rfishbase::country()` although they occur all around Timor and Indonesia.
https://www.fishbase.se/summary/Hyporhamphus-quoyi.html
https://www.fishbase.se/summary/Abudefduf-vaigiensis.html

Reply:
Thanks for checking this too! It might be worth to add a comment in the code to explain why is this the case.